src/labone/nodetree/node.py

```diff
@@ -0,0 +1,1038 @@
+                f"is invalid, because {current_node.path} "
+                f"is already a leaf-node."
+            )
+            raise RuntimeError(msg) from e
+
+        return current_node
+
+    def _try_generate_subnode(
+        self,
+        next_path_segment: NormalizedPathSegment,
+    ) -> ResultNode | AnnotatedValue:
+        """Provides nodes for the extended path and the original values for leafs."""
+        extended_path = (*self._path_segments, next_path_segment)
+
+        try:
+            next_subtree: (
+                list[list[NormalizedPathSegment]]
+                | NestedDict[list[list[NormalizedPathSegment]] | dict]
+            ) = self._subtree_structure[next_path_segment]
+        except KeyError as e:
+            if next_path_segment == WILDCARD:
+                msg = (
+                    "Wildcards '*' are not allowed in a tree representing "
+                    "measurement results"
+                )
+                raise RuntimeError(msg) from e
+
+            msg = (
+                f"node path_segments '{self}' cannot be extended with "
+                f"/{next_path_segment}\nviable extensions would be "
+                f"{list(self._subtree_structure.keys())}"
+            )
+            raise RuntimeError(msg) from e
+        else:
+            deeper_node = ResultNode(
+                extended_path,
+                self._subtree_structure[next_path_segment],
+                self._value_structure,
+                self._timestamp,
+                self._path_aliases,
+            )
+
+            if not next_subtree:
+                # give value instead of subnode if already at a leaf
+                # generate hypothetical node in order to apply normal behaviour,
+                # including path alias redirection
+                return self._value_structure[deeper_node.path]
+
+            return deeper_node
+
+    def __dir__(self) -> t.Iterable[str]:
+        return [pythonify_path_segment(p) for p in self._subtree_structure]
+
+    def __contains__(self, item: str | int | ResultNode | AnnotatedValue) -> bool:
+        """Checks if a path-segment or node is an immediate sub-node of this one."""
+        if isinstance(item, ResultNode):
+            return self.is_child_node(item)
+        if isinstance(item, AnnotatedValue):
+            return self.is_child_node(split_path(item.path))
+        return normalize_path_segment(item) in self._subtree_structure
+
+    def __call__(self, *_, **__) -> None:
+        """Do not use. Showing an error to express result-nodes can't be get/set."""
+        msg = (
+            "Trying to get/set on a result node, which is not possible. "
+            "This node only represents values of a former get-request. "
+            "To interact with a device, make sure to operate on normal nodes."
+        )
+        raise RuntimeError(msg)
+
+    def __repr__(self) -> str:
+        return f"Results for {self.path} at timestamp {self._timestamp}"
```

Comment:
still not fixed. The repr and str should print all available results (similar to a dict)

Reply:
It would be fine for me ... but if you want we can keep the repr as it is, but in the str repr we MUST print all results ... it would be super annoying if you get nodes and print the result and its not showing the result